src/index.ts

```diff
@@ -44,7 +44,7 @@ export function sync(
   // sync router on store change
   const storeUnwatch = store.watch(
```

Comment:
The `{ sync: true }` below should be changed to `{ flush: 'sync' }` ([Vue 2](https://github.com/vuejs/vue/blob/b51430f598b354ed60851bb62885539bd25de3d8/src/core/observer/watcher.js#L34) vs [Vue 3](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/apiWatch.ts#L58))

Reply:
Thank you! 
I fixed it!